controllers/C_Document.class.php

```diff
@@ -801,18 +796,20 @@ public function retrieve_action(string $patient_id = null, $document_id, $as_fil
         if (file_exists($temp_url)) {
             $url = $temp_url;
         }
-        //fire a remote call to see if the file is stored somewhere else
-        $s3Key = explode("//", $temp_url); //split the url to get the s3 key
-        // if the s3 key isn't empty then ...
-        if (!empty($s3Key[1])) {
```

Comment:
if (!empty($s3Key[1])) {

This would prevent the event from being fired if the file name is empty. Now, I moved that logic into the module. It sends and empty return if there is no file name connected to the URL. i.e.  " C:/xampp/htdocs/openemr/sites/default/documents// " occurs. 

    if (!file_exists($url)) {

This logic should keep the system from falling through. It should exit because the file won't exist on the local drive.